src/java/htsjdk/samtools/SAMTextWriter.java

```diff
@@ -93,7 +154,19 @@ public void writeAlignment(final SAMRecord alignment) {
         try {
             out.write(alignment.getReadName());
             out.write(FIELD_SEPARATOR);
-            out.write(Integer.toString(alignment.getFlags()));
+            switch (this.flagFieldsOutput) {
```

Comment:
Warning: quite possibly more trouble than it's worth, but this could be converted to polymorphism in the enum instances:

```
public enum FlagFieldsOutput {
  ...
  HEXADECIMAL {
    @Override
    public String format(int flags) {
      return String.format("0x%x", alignment.getFlags());
    }
    public abstract String format(int flags);
  },
  ...
}
```

Then this code becomes:

```
  out.write(flagFieldsOutput.format(alignment.getFlags()));
```